lib/simctl.js

```diff
@@ -75,7 +75,11 @@ async function createDevice (name:string, deviceTypeId:string,
 }
 
 async function deleteDevice (udid:string):void {
-  await simExec('delete', 0, [udid]);
+  let loopFn:Function = async () => {
+    await simExec('delete', 0, [udid]);
+  };
+  // retry delete with a sleep in between because it's flakey
+  await retryInterval(5, 200, loopFn);
```

Comment:
Did you check the contents of stderr when the exit code was `1`? I'm thinking maybe we tried to delete a sim which was either still running, or already deleted.

I'm wary of these retries that get shot-off, I think they lead to hidden issues and performance hits.

Maybe make a separate `deleteWithRetry` method or something? Then you can use that in your tests and not every time we use this function? I'm running lots of tests that delete sims in appium-webdriveragent-driver without having to retry.